torch/csrc/jit/python/init.cpp

```diff
@@ -885,13 +886,107 @@ void initJITBindings(PyObject* module) {
+          // provide a pointer, which points to the original method, i.e.,
+          // itself. py::cpp_function doesn't support specifying the function
+          // signature without real implementation, so we we use a trick here.
+          // First, we create a function holder pointing to an empty
+          // cpp_function. Then we use a reference as the alias of the empty
+          // function. Then we define the real returned function, and update
+          // the value of the function pointed by the holder using the alias.
+          // With this trick, we are able to call self inside the returned
+          // function.
+          auto func_holder = std::make_unique<py::cpp_function>();
+          py::cpp_function& func_ref = *func_holder;
+          auto func = py::cpp_function(
+              [operations, op_name, self_ptr = std::move(func_holder)](
+                  py::args args, py::kwargs kwargs) {
+                std::shared_ptr<Operator> found_op = nullptr;
+                if (operations.size() == 1) {
+                  found_op = operations.at(0);
+                } else {
+                  for (const auto& op : operations) {
+                    const auto& expected_args = op->schema().arguments();
+                    bool match = true;
+                    for (size_t i = args.size(); i < expected_args.size();
+                         ++i) {
+                      if (!kwargs.contains(expected_args[i].name().c_str()) &&
+                          !expected_args[i].default_value().has_value()) {
+                        match = false;
+                        break;
+                      }
+                    }
+                    if (match) {
+                      found_op = op;
+                      break;
+                    }
+                  }
+                }
+                std::vector<py::handle> overloaded_args;
+                if (found_op) {
+                  const auto& expected_args = found_op->schema().arguments();
+                  for (size_t i = 0; i < expected_args.size(); ++i) {
+                    if (i < args.size()) {
+                      if (THPVariable_CheckExact(args[i].ptr())) {
+                        continue;
+                      }
+                      if (check_has_torch_function(args[i].ptr())) {
+                        append_overloaded_arg(overloaded_args, args[i].ptr());
+                      }
+                    } else if (kwargs.contains(
+                                   expected_args[i].name().c_str())) {
+                      const auto& kwarg =
+                          kwargs[expected_args[i].name().c_str()];
+                      if (THPVariable_CheckExact(kwarg.ptr())) {
+                        continue;
+                      }
+                      if (check_has_torch_function(kwarg.ptr())) {
+                        append_overloaded_arg(overloaded_args, kwarg.ptr());
+                      }
```

Comment:
This logic is flawed and will cause the same issue as in #34294. We need a way to recurse into `list`s and `tuple`s.